lib/elastomer/client/docs.rb

```diff
@@ -225,8 +225,11 @@ def search(query, params = nil)
       # See https://www.elastic.co/guide/en/elasticsearch/reference/current/search-shards.html
       #
       # Returns the response body as a hash
-      def search_shards(params = {})
-        response = client.get "/{index}{/type}/_search_shards", update_params(params, action: "docs.search_shards", rest_api: "search_shards")
+      def search_shards(params = {}, remove_type_param = false)
```

Comment:
The update_params method adds the type param to each request. For ES 7 it adds the default type `_doc`. For most endpoints, adding `_doc` works fine for ES 7, but for `_search_shards`, `index/type/_search_shards` does not return the right result. So I'm removing the type param only for this method. This is not the best approach, so I'm open to suggestions for a better way to handle this (I couldn't find a usage for this endpoint on dotcom so this change shouldn't actually affect anything)

Reply:
Based on our discussion today, I think this is the way to go. Even after we handle defaulting the `type` for ES 7 in `update_params`, this one would still get a `type` so removing it here seems good for now.